flake.nix

```diff
@@ -0,0 +1,26 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }: rec {
+    lib = {
+      buildTerranix = { pkgs, terranix_config, ... }@terranix_args: let
+        terraform = import ./core/default.nix terranix_args;
+        config_json = pkgs.writeTextFile {
+          name = "terraform-config";
+          text = builtins.toJSON terraform.config;
+          executable = false;
+          destination = "/config.tf.json";
+        };
+      in
+        config_json;
```

Comment:
Hi! I am a fan of this project and [use it with flakes](https://github.com/NEU-DSG/dailp-encoding/blob/main/flake.nix#L80). This looks really similar to how I use it and the tips in #9, I might just collapse the `config_json` variable into the body expression here. I'd love to see this merged to bring terranix into the flaky future!

Reply:
Hey @loafofpiecrust nice to hear. Sadly I haven't had the time to read into flakes so far. But I will definitely put flakes into terranix in 2021.